### PR TITLE
Bugfix/cover storage path

### PR DESCRIPTION
**Pull request by sunnylochun**

### CODE REVIEW

**Comment by sunnylochun** [comment]
[(3) 【部落格】無法替換封面圖片](https://trello.com/c/cUXbh4lh/2986-3-%E3%80%90%E9%83%A8%E8%90%BD%E6%A0%BC%E3%80%91%E7%84%A1%E6%B3%95%E6%9B%BF%E6%8F%9B%E5%B0%81%E9%9D%A2%E5%9C%96%E7%89%87)

File: src/components/activity/ActivityIntroductionForm.tsx [hunk 1]
```diff
@@ -34,6 +35,7 @@ const ActivityIntroductionForm: React.FC<{
     hasura.UPDATE_ACTIVITY_INTRODUCTIONVariables
   >(UPDATE_ACTIVITY_INTRODUCTION)
   const [loading, setLoading] = useState(false)
+  const [coverId, setCoverId] = useState(uuid())
```
Comment (kkshyu) [inline src/components/activity/ActivityIntroductionForm.tsx:38]: i think we may not need to use the state here. how about just use `const coverId = uuid()` directly?

File: src/components/blog/BlogPostSettingForm.tsx [hunk 2]
```diff
@@ -31,22 +32,23 @@ const BlogPostSettingForm: React.FC<{
     hasura.UPDATE_POST_MERCHANDISE_COLLECTIONVariables
   >(UPDATE_POST_MERCHANDISE_COLLECTION)
   const [loading, setLoading] = useState(false)
+  const [coverId, setCoverId] = useState(uuid())
```
Comment (kkshyu) [inline src/components/blog/BlogPostSettingForm.tsx:35]: the same as the above

File: src/components/merchandise/MemberShopBasicForm.tsx [hunk 3]
```diff
@@ -35,6 +36,7 @@ const MemberShopBasicForm: React.FC<{
     hasura.UPDATE_MEMBER_SHOP_COVERVariables
   >(UPDATE_MEMBER_SHOP_COVER)
   const [loading, setLoading] = useState(false)
+  const [coverId, setCoverId] = useState(uuid())
```
Comment (kkshyu) [inline src/components/merchandise/MemberShopBasicForm.tsx:39]: the same as the above

File: src/components/podcast/PodcastProgramIntroForm.tsx [hunk 4]
```diff
@@ -25,6 +26,7 @@ const PodcastProgramIntroForm: React.FC<{
   const [form] = useForm<FieldProps>()
   const { id: appId } = useApp()
   const [loading, setLoading] = useState(false)
+  const [coverId, setCoverId] = useState(uuid())
```
Comment (kkshyu) [inline src/components/podcast/PodcastProgramIntroForm.tsx:29]: the same as the above

File: src/components/project/ProjectIntroForm.tsx [hunk 5]
```diff
@@ -64,15 +66,16 @@ const ProjectIntroForm: React.FC<{
     updateProjectCover({
       variables: {
         projectId: project.id,
-        previewUrl: `https://${process.env.REACT_APP_S3_BUCKET}/project_covers/${appId}/${project.id}?t=${uploadTime}`,
+        previewUrl: `https://${process.env.REACT_APP_S3_BUCKET}/project_covers/${appId}/${project.id}/${coverId}?t=${uploadTime}`,
         coverUrl:
           project.coverUrl === null
-            ? `https://${process.env.REACT_APP_S3_BUCKET}/project_covers/${appId}/${project.id}?t=${uploadTime}`
+            ? `https://${process.env.REACT_APP_S3_BUCKET}/project_covers/${appId}/${project.id}/${coverId}?t=${uploadTime}`
```
Comment (kkshyu) [inline src/components/project/ProjectIntroForm.tsx:72]: we don't need the `uploadTime` if we have used the `uuid`

File: src/pages/ProgramAdminPage/ProgramIntroForm.tsx [hunk 6]
```diff
@@ -47,11 +49,12 @@ const ProgramIntroForm: React.FC<{
     updateProgramCover({
       variables: {
         programId: program.id,
-        coverUrl: `https://${process.env.REACT_APP_S3_BUCKET}/program_covers/${appId}/${program.id}?t=${uploadTime}`,
+        coverUrl: `https://${process.env.REACT_APP_S3_BUCKET}/program_covers/${appId}/${program.id}/${coverId}?t=${uploadTime}`,
```
Comment (kkshyu) [inline src/pages/ProgramAdminPage/ProgramIntroForm.tsx:52]: the same as the above, no need `uploadTime`

**Review by kkshyu — CHANGES_REQUESTED** [review]

File: src/components/activity/ActivityIntroductionForm.tsx (hunk 1 above)
Comment (sunnylochun) [reply src/components/activity/ActivityIntroductionForm.tsx]: fixed

File: src/components/project/ProjectIntroForm.tsx (hunk 5 above)
Comment (sunnylochun) [reply src/components/project/ProjectIntroForm.tsx]: fixed

**Comment by emily40830** [comment]
please solve the conflict QQ

**Review by emily40830 — APPROVED** [review]

**Pull request merged**